internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -695,10 +739,143 @@ func NewOpenShiftAuthProxyContainer(cr *model.CryostatInstance, specs *ServiceSp
 	}
 }
 
-// func NewOAuth2ProxyContainer(cr *model.CryostatInstance, specs *ServiceSpecs, imageTag string,
-// tls *TLSConfig) corev1.Container {
+func isOpenShiftAuthProxyDisabled(cr *model.CryostatInstance) bool {
+	if cr.Spec.AuthorizationOptions != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.Disabled != nil {
+		return *cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.Disabled
+	}
+	return false
+}
+
+func getOpenShiftSubjectAccessReviews(cr *model.CryostatInstance) []authzv1.ResourceAttributes {
+	if cr.Spec.AuthorizationOptions != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.SubjectAccessReview != nil {
+		return []authzv1.ResourceAttributes{*cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.SubjectAccessReview}
+	}
+	return []authzv1.ResourceAttributes{
+		getDefaultOpenShiftAccessRole(cr),
+	}
+}
+
+func getOpenShiftTokenReview(cr *model.CryostatInstance) authzv1.ResourceAttributes {
+	if cr.Spec.AuthorizationOptions != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles != nil && cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.TokenReview != nil {
+		return *cr.Spec.AuthorizationOptions.OpenShiftSsoRoles.TokenReview
+	}
+	return getDefaultOpenShiftAccessRole(cr)
+}
+
+func getDefaultOpenShiftAccessRole(cr *model.CryostatInstance) authzv1.ResourceAttributes {
```

Comment:
I guess I can replace this function and some of the logic around it by using `internal/webhooks/defaulter.go`, right?

Reply:
You could, but I think for the most part we still don't set defaults in the CR, and just do like you've done already here. It could cause confusion for the user if we set these fields when running on Kubernetes for example.